### PR TITLE
Check Result Filter

### PR DESCRIPTION
**Pull request by CDimonaco**
We need to implement a filter on the check results page.

This filter will contain the options for filtering based on check result, passing, warning, critical etc..

The result will match the provided mockups

### CODE REVIEW

**Comment by CDimonaco** [comment]
This is the result

trento_check_filtering

**Review by arbulu89 — APPROVED** [review]
Great!
The code is a bit tricky to understand, but it works great.
The unique thing that I could highlight is that in the backend we have the `unknown` option as well, so maybe we should have it here as well

Besides that, I would personally prefer to have the result icons in the filter rather than the text. What do you think on this @jagabomb ? Or at least some visual hint

File: assets/js/components/ClusterDetails/ChecksResults.jsx [hunk 1]
```diff
@@ -234,24 +240,32 @@ export const ChecksResults = () => {
...
-            className="w-1/3 bg-waterhole-blue text-white"
+            className="w-2/3 text-jungle-green-500 text-left py-0 px-0"
             size="small"
+            type="transparent"
             onClick={() => navigate(`/clusters/${cluster.id}`)}
           >
-            <EOS_ARROW_BACK className="inline-block fill-white" /> Back to
-            Cluster
+            <EOS_ARROW_BACK className="inline-block fill-jungle-green-500" />{' '}
+            Back to Cluster Detail
```
[the remark this answers is not in the document]
Comment (jagabomb) [reply assets/js/components/ClusterDetails/ChecksResults.jsx:252]: My bad in the UI. This is a nicer convention to call it **Back to Cluster Details**.

**Review by dottorblaster — APPROVED** [review]
LGTM, the custom hook was a little too fancy for me, but it does no harm and I want to respect your creativity. Also, the multi-field-filtering thing is a nice thought.

File: assets/js/components/ClusterDetails/ChecksResults.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (dottorblaster) [reply assets/js/components/ClusterDetails/ChecksResults.jsx:252]: @jagabomb well you can't tell @CDimonaco doesn't follow your directions :smile:

**Comment by jagabomb** [comment]
> Besides that, I would personally prefer to have the result icons in the filter rather than the text. What do you think on this @jagabomb ? Or at least some visual hint

@arbulu89 Maybe this is an enhancement on how dropdown can be more visual/pretty? Reason I say this is that this screen is one example of this Health Check dropdown component. So if we change it here then we will have to re-look/change it on other screens for consistency.

File: assets/js/components/ClusterDetails/ChecksResults.jsx [hunk 2]
```diff
@@ -23,9 +23,11 @@ import {
 } from './ClusterDetails';
 import { ExecutionIcon } from './ExecutionIcon';
 import { getClusterName } from '@components/ClusterLink';
+import ChecksResultFilters from '@components/ClusterDetails/ChecksResultFilters';
```
[the remark this answers is not in the document]
Comment (CDimonaco) [reply assets/js/components/ClusterDetails/ChecksResults.jsx:26]: I'll fix it, the linter should catch it 👀

**Comment by arbulu89** [comment]
> > Besides that, I would personally prefer to have the result icons in the filter rather than the text. What do you think on this @jagabomb ? Or at least some visual hint
> 
> @arbulu89 Maybe this is an enhancement on how dropdown can be more visual/pretty? Reason I say this is that this screen is one example of this Health Check dropdown component. So if we change it here then we will have to re-look/change it on other screens for consistency.

Yeah, maybe at some point we have reuse the results/state filtering box in all the views. Maybe work for the future hehe

**Pull request closed without merging**